### PR TITLE
Initial checkin

### PR DESCRIPTION
**Pull request by adampauls**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,33 +1,168 @@
...
+print(tensor)
+```
+
+Scala-Torch:
+```scala
+val tensor = torch.ones($(4, 4))
+println(s"First row: ${tensor(0)}")
+println(s"First column: ${tensor(::, 0)}")
+println(s"Last column: ${tensor(---, -1)}")
+tensor(::, 1) = 0
```
Comment (sammthomson) [inline README.md:62]: just to match the python example
```suggestion
tensor(::, 1) = 0
println(tensor)
```

File: README.md [hunk 2]
```diff
@@ -1,33 +1,168 @@
...
+See [this file](scala-torch/src/test/scala/com/microsoft/scalatorch/torch/tutorial/PyTorchOrgTensorTutorialTest.scala) for 
+a complete translation of the PyTorch tutorial into Scala-Torch.
+
+### Memory management
+
+One big difference between Scala-Torch and PyTorch is in memory management. Because Python and LibTorch both use 
+reference counting, memory management is fairly transparent to users. However, since the JVM uses garbage collection
+and [finalizers are not guaranteed to run](https://docs.oracle.com/javase/9/docs/api/java/lang/Object.html#finalize--),
+it is not easy to make memory management transparent to the user. Scala-Torch elects to make memory managment something
+the user most control by providing [ReferenceManager](scala-torch/src/main/scala/com/microsoft/scalatorch/torch/ReferenceManager.scala)s
```
Comment (sammthomson) [inline README.md]:
```suggestion
the user must control by providing [ReferenceManager](scala-torch/src/main/scala/com/microsoft/scalatorch/torch/ReferenceManager.scala)s 
```

**Review by sammthomson — APPROVED** [review]
🎉

**Comment by ctongfei** [comment]
🎉

**Review by ctongfei — APPROVED** [review]

**Pull request closed without merging**